clients/src/main/java/org/apache/kafka/common/requests/CreateAclsRequest.java

```diff
@@ -51,9 +52,20 @@
                     PERMISSION_TYPE))));
 
     /**
-     * The version number is bumped to indicate that on quota violation brokers send out responses before throttling.
+     * V1 sees a new `RESOURCE_NAME_TYPE` that controls how the resource name is interpreted and version
+     * was bumped to indicate that, on quota violation, brokers send out responses before throttling.
+     *
+     * For more info, see {@link org.apache.kafka.common.resource.ResourceNameType}.
      */
-    private static final Schema CREATE_ACLS_REQUEST_V1 = CREATE_ACLS_REQUEST_V0;
+    private static final Schema CREATE_ACLS_REQUEST_V1 = new Schema(
+            new Field(CREATIONS_KEY_NAME, new ArrayOf(new Schema(
+                    RESOURCE_TYPE,
+                    RESOURCE_NAME,
+                    RESOURCE_NAME_TYPE,
+                    PRINCIPAL,
+                    HOST,
+                    OPERATION,
+                    PERMISSION_TYPE))));
 
```

Comment:
I can't figure out how to add a comment to the correct line using github's terrible user interface.  But on line 110 where you have the code:

```
        @Override
        public CreateAclsRequest build(short version) {
            return new CreateAclsRequest(version, creations);
        }
```
You must add verification that the version you are using supports the ACLs you are trying to send.  If someone tries to set a prefix ACL when using CREATE_ACLS_REQUEST_V0, the code must throw `UnsupportedVersionException`.  In fact, anything besides `ResourceNameType.LITERAL` should trigger a UVE there.

Reply:
Consider it done.